subsys/testsuite/ztest/src/ztest.c

```diff
@@ -1321,15 +1326,57 @@ static int cmd_shuffle(const struct shell *sh, size_t argc, char **argv)
 
 static int cmd_run_suite(const struct shell *sh, size_t argc, char **argv)
 {
+	struct getopt_state *state;
+	int opt;
+	static struct option long_options[] = {{"repeat_iter", required_argument, 0, 'r'},
+		{0, 0, 0, 0}};
+	int opt_index = 0;
+	int val;
+	int opt_num = 0;
+	void *param = NULL;
+	int repeat_iter = 1;
+
+	while ((opt = getopt_long(argc, argv, "r:p:", long_options, &opt_index)) != -1) {
+		state = getopt_state_get();
+		switch (opt) {
+		case 'r':
+			val = atoi(state->optarg);
+			if (val < 1) {
+				shell_fprintf(sh, SHELL_ERROR,
+					"Invalid number of suite interations\n");
+				return -ENOEXEC;
+			}
+			repeat_iter = val;
+			opt_num++;
+			break;
+		case 'p':
+			param = state->optarg;
+			opt_num++;
+			break;
```

Comment:
It seems that current solution allows for only one parameter. That parameter is CLI-provided, so it is in practice limited to simple types like `char*`, `int`, `float`. What if I need multiple parameters? I don't think I can easily pass in a struct-like structure and neither can I use multiple `-p` flags, correct?

Is there a method I have overlooked, or is this a possible enhancement?

Reply:
So far, I've provided enhancement for passing one parameter into instance of the test.
If necessary, I'll expand this functionality for passing more complex types of parameters or multiple parameters.
In this approach we're based on the shell. So, the valid question is, how would you like to pass more complex parameters via shell?

The last test case explains how to pass data set into the test. But you're right it's technically one parameter for one instance of the test.

Reply:
PS: I'd like to avoid passing the hardcoded parameters (the parameters which are the part of the source of a test) . But technically you are able to pass more complex parameters using the fixtures in ztest. Please take a look at https://docs.zephyrproject.org/latest/develop/test/ztest.html#test-fixtures

Reply:
> So far, I've provided enhancement for passing one parameter into instance of the test. If necessary, I'll expand this functionality for passing more complex types of parameters or multiple parameters. In this approach we're based on the shell. So, the valid question is, how would you like to pass more complex parameters via shell?
> 
> The last test case explains how to pass data set into the test. But you're right it's technically one parameter for one instance of the test.

One practical method, in my opinion, would be something like
`<ztest call> -p param1=param1value -p param2=param2value`
that then gets transformed into a `data` struct with two fields, `param1` and `param2` that have the desired values. `zassert_equal(data->param1, "param1value")` should pass. List-like solution (think Twister's `--platform` option) seems incorrect here.

However, there are probably many other approaches. I personally think that expansion of this solution to multiple parametrisation would best be handled in a separate PR. That's why I gave an approve.

Reply:
I think that's quite complex, because while initializing the sub-command via `SHELL_CMD_ARG` you are informing how many optional arguments this particular sub-command is expecting.